### PR TITLE
PVN-10 Factorize mailbox EntityManagerFactory creation (JPA)

### PR DESCRIPTION
**Pull request by chibenwa**
Easy win for JPA information duplication...

### CODE REVIEW

File: mailbox/jpa/src/test/java/org/apache/james/mailbox/jpa/EntityManagerFactorySupplier.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            JPAMailboxMessage.class.getName() + ";" +
+            JPAProperty.class.getName() + ";" +
+            JPAUserFlag.class.getName() + ";" +
+            JPAMailboxAnnotation.class.getName() + ";" +
+            JPASubscription.class.getName() + ")");
+
+        return OpenJPAPersistence.getEntityManagerFactory(properties);
+    }
+
+    public void clear() {
```
Comment (rouazana) [inline mailbox/jpa/src/test/java/org/apache/james/mailbox/jpa/EntityManagerFactorySupplier.java]: What does mean clear for a supplier?
It seems you are mixing responsibilities.

File: mailbox/jpa/src/test/java/org/apache/james/mailbox/jpa/EntityManagerFactorySupplier.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import org.apache.james.mailbox.jpa.mail.model.JPAProperty;
+import org.apache.james.mailbox.jpa.mail.model.JPAUserFlag;
+import org.apache.james.mailbox.jpa.mail.model.openjpa.AbstractJPAMailboxMessage;
+import org.apache.james.mailbox.jpa.mail.model.openjpa.JPAMailboxMessage;
+import org.apache.james.mailbox.jpa.user.model.JPASubscription;
+import org.apache.openjpa.persistence.OpenJPAPersistence;
+
+import com.google.common.base.Supplier;
+
+public class EntityManagerFactorySupplier implements Supplier<EntityManagerFactory> {
```
Comment (aduprat) [inline mailbox/jpa/src/test/java/org/apache/james/mailbox/jpa/EntityManagerFactorySupplier.java]: Buzzword winner !

Comment (chibenwa) [inline mailbox/jpa/src/test/java/org/apache/james/mailbox/jpa/EntityManagerFactorySupplier.java]: This class no more exists, cf Raphael previous comment

**Comment by chibenwa** [comment]
```
[Building Java 8 pvn-10-proposal-8fb1bc34-c30f-4a41-9ac7-822e63e4cf1c] Failed tests: 
[Building Java 8 pvn-10-proposal-8fb1bc34-c30f-4a41-9ac7-822e63e4cf1c]   expected:<[200]> but was:<[404]>
[Building Java 8 pvn-10-proposal-8fb1bc34-c30f-4a41-9ac7-822e63e4cf1c]   expected:<[200]> but was:<[404]>
[Building Java 8 pvn-10-proposal-8fb1bc34-c30f-4a41-9ac7-822e63e4cf1c] 
[Building Java 8 pvn-10-proposal-8fb1bc34-c30f-4a41-9ac7-822e63e4cf1c] Tests run: 628, Failures: 2, Errors: 0, Skipped: 2
```

Seems that I am not lucky with CI...

**Comment by chibenwa** [comment]
test this please

**Pull request merged**